src/Reflection/ReflectionFunctionAbstract.php

```diff
@@ -97,13 +101,17 @@ public function getNumberOfParameters()
      */
     public function getNumberOfRequiredParameters()
     {
-        return count(array_filter($this->parameters, function (ReflectionParameter $p) {
-            return !$p->isOptional();
-        }));
+        return count(array_filter(
```

Comment:
I had an idea to use this `getNumberOfRequiredParameters` to determine `isOptional`, until I discovered that that would cause cyclic dependency... which was annoying :(

Can always create an issue to resolve this later, but it works